File: 646-MaximumLengthOfPairChain/646-MaximumLengthOfPairChain.cpp
// Last updated: 6/11/2026, 11:32:12 AM
class Solution {
public:
    int findLongestChain(vector<vector<int>>& pairs) {
        int prev = -1;

        sort(pairs.begin(), pairs.end(), [](vector<int> &a, vector<int> &b){
            return a[1]<b[1];
        });
        
        int n = pairs.size();
        int cur = 0;
        int cnt = 0;
        while(cur<n){
            if(prev==-1 || (pairs[cur][0]>pairs[prev][1])){
                cnt++;
                prev = cur;
                cur++;
            }else{
                cur++;
            }
        }

        return cnt;


    }
};

File: 646-MaximumLengthOfPairChain/646-MaximumLengthOfPairChain.py
# Last updated: 6/11/2026, 11:32:18 AM
class Solution:
    def findLongestChain(self, pairs: List[List[int]]) -> int:
        pairs = sorted(pairs, key = lambda x: x[1])
        cnt = 0
        prevtail = -10000
        for head, tail in pairs:
            if(head>prevtail):
                prevtail = tail
                cnt+=1
        return cnt